xml/net/server/protocol.xml

```diff
@@ -2242,17 +2239,17 @@
 
     <packet family="Trade" action="Reply">
         <comment>Trade updated (items changed)</comment>
-        <field name="trade_data" type="TradeItemData"/>
+        <array name="data" type="TradeItemData" length="2"/>
```

Comment:
Instead of inlining an array to each packet and changing the granularity of the `TradeItemData` struct, what about changing `TradeItemData` to have a fixed-size array of `PlayerTradeItemData`?
That would be less disruptive in terms of API modeling in eolib implementations, and it would also reduce this repetition of the array in all the packets.
Thoughts?

Reply:
I think in practice the extra level of indirection makes EOLib implementations more annoying to use (having to create deep object hierarchies just to send a packet, since applications usually store the data in their own format). EOLib sits at the presentation layer so I think simplifying that code required to translate data from the application to presenter layer is more valuable than a slight disruption in the API.

Additionally, there is an aspect of inconsistency since there is no precedent for encapsulation of fixed size arrays in the rest of the API (pub rids, citizen questions/answers, paperdoll rings/armlets/bracers, etc.).

Reply:
Compelling argument - alright, I'm happy to stick with the inlined array fields.